src/util/internalCallTree.js

```diff
@@ -88,7 +88,15 @@ class InternalCallTree {
 async function buildTree (tree, step, scopeId) {
   let subScope = 1
   tree.scopeStarts[step] = scopeId
-  tree.scopes[scopeId] = { firstStep: step, locals: {} }
+  tree.scopes[scopeId] = { firstStep: step, locals: {} } 
+
+  function checkDepth (step, trace) {
```

Comment:
Please rename such that the nature of the check is visible from the name, for example `changesDepth`. The depth of what is it actually?